client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -40,8 +40,7 @@ object Retry {
           policy(req, Left(e), attempts) match {
             case Some(duration) =>
               // info instead of error(e), because e is not discarded
-              logger.info(
-                s"Request $req threw an exception on attempt #$attempts with message: ${e.getMessage}. Retrying after $duration.")
+              logger.info(e)(s"Request threw an exception on attempt #$attempts. Retrying after $duration")
```

Comment:
Would change the line above, too: log `e` because it's discarded, but at info level because we're trying again.

Reply:
Good point. I'll change it too. Could you please explain:

> log e because it's discarded

? 

I don't understand this statement.

Thanks